### PR TITLE
Improved unit test coverage.

### PR DESCRIPTION
**Pull request by balajismaniam**
Relates to #62. Coverage is now at 64%.
- Refactored code for testing purposes.
- Misc. comment changes.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -148,21 +184,26 @@ func main() {
 			stdoutLogger.Printf("%s = %s", name, value)
 			// Skip if label doesn't match labelWhiteList
 			if !labelWhiteList.Match([]byte(name)) {
-				stderrLogger.Printf("%s does not match the whitelist (%s) and will not be published.", name, whiteListArg)
+				stderrLogger.Printf("%s does not match the whitelist and will not be published.", name)
```
Comment (ConnorDoyle) [inline main.go]: We should retain the old behavior of printing the whitelist regular expression to help with debugging. We could still stringify `labelWhitelist`) (https://golang.org/pkg/regexp/#Regexp.String).

File: main_test.go [hunk 2]
```diff
@@ -103,6 +116,151 @@ func TestDiscoveryWithMockSources(t *testing.T) {
...
+			Convey("--no-publish is set and sourcesArg is set to appropriate values", func() {
+				So(noPublish, ShouldBeTrue)
+				So(sourcesArg, ShouldResemble, []string{"fake1", "fake2", "fake3"})
+				So(len(whiteListArg), ShouldEqual, 0)
+			})
+		})
+	})
+}
+
+func TestConfigureParametes(t *testing.T) {
```
Comment (ConnorDoyle) [inline main_test.go]: nit: s/TestConfigureParametes/TestConfigureParameters

**Comment by balajismaniam** [comment]
@ConnorDoyle PTAL. I updated it.

**Comment by balajismaniam** [comment]
I don't think the bot picked-up the comment from your approval. I will self lgtm.

**Pull request merged**